presto-orc/src/main/java/com/facebook/presto/orc/metadata/OrcFileTail.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.orc.metadata;
+
+import com.facebook.presto.orc.OrcDecompressor;
+import com.facebook.presto.orc.metadata.PostScript.HiveWriterVersion;
+
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+
+public class OrcFileTail
+{
+    private final HiveWriterVersion hiveWriterVersion;
+    private final int bufferSize;
+    private final CompressionKind compressionKind;
+    private final Optional<OrcDecompressor> decompressor;
```

Comment:
We don't have to cache this. This will create a buffer inside the decompressor. A decompressor can be recreated with bufferSize and compressionKind.